### PR TITLE
Fixes #17128 .

### LINKED ISSUE #17128: fixing VisibleDeprecationWarning
### System Info

```shell
- `transformers` version: 4.18.0
- Platform: Linux-5.17.1-arch1-1-x86_64-with-glibc2.35
- Python version: 3.9.7
- Huggingface_hub version: 0.5.1
- PyTorch version (GPU?): 1.10.2+cu102 (True)
- Tensorflow version (GPU?): 2.8.0 (True)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>
```

### Who can help?

Hi @LysandreJik ,
    
Could you please fix `VisibleDeprecationWarning` warning? 

Based on [this stackoverflow thread](https://stackoverflow.com/q/63097829), specifying `dtype=object` fixes the issue.

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Code

```
from transformers import pipeline

question_answerer = pipeline(task="question-answering", model="distilbert-base-cased-distilled-squad")

context = """
Captain America, and Wonder Woman.
Between 1939 and 1941 Detective Comics and its sister company, All-American Publications, 
introduced popular superheroes such as Batman and Robin, Wonder Woman, the Flash, 
Green Lantern, Doctor Fate, the Atom, Hawkman, Green Arrow and Aquaman.[7] Timely Comics, 
the 1940s predecessor of Marvel Comics, had million-selling titles featuring the Human Torch,
the Sub-Mariner, and Captain America.[8]
As comic books grew in popularity, publishers began launching titles that expanded 
into a variety of genres. Dell Comics' non-superhero characters (particularly the 
licensed Walt Disney animated-character comics) outsold the superhero comics of the day.[12] 
The publisher featured licensed movie and literary characters such as Mickey Mouse, Donald Duck,
Roy Rogers and Tarzan.[13] It was during this era that noted Donald Duck writer-artist
Carl Barks rose to prominence.[14] Additionally, MLJ's introduction of Archie Andrews
in Pep Comics #22 (December 1941) gave rise to teen humor comics,[15] with the Archie 
Andrews character remaining in print well into the 21st century.[16]
At the same time in Canada, American comic books were prohibited importation under 
the War Exchange Conservation Act[17] which restricted the importation of non-essential 
goods. As a result, a domestic publishing industry flourished during the duration 
of the war which were collectively informally called the Canadian Whites.
The educational comic book Dagwood Splits the Atom used characters from the comic 
strip Blondie.[18] According to historian Michael A. Amundson, appealing comic-book 
characters helped ease young readers' fear of nuclear war and neutralize anxiety 
about the questions posed by atomic power.[19] It was during this period that long-running 
humor comics debuted, including EC's Mad and Carl Barks' Uncle Scrooge in Dell's Four 
Color Comics (both in 1952).[20][21]
"""

question = "What popular superheroes were introduced between 1939 and 1941?"
result = question_answerer(question=question, context=context)

print(result['answer'])
```

### Output

```
/home/myuser/anaconda3/lib/python3.9/site-packages/numpy/core/_asarray.py:102: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
  return array(a, dtype, copy=False, order=order)
EC's Mad and Carl Barks' Uncle Scrooge
```

### Expected behavior

```shell
Warning should not be generated. Accoring to [this post](https://forums.fast.ai/t/visibledeprecationwarning-creating-an-ndarray-from-ragged-nested-sequences-is-deprecated/81774/3), this warning is likely to turn into an error in future releases.
```

### PR DESCRIPTION
**Pull request by mygithubid1**
# What does this PR do?
Finalize changes as per this [PR](https://github.com/huggingface/transformers/pull/17277).
Sorry. I messed up changes during merge and hence this new PR.

Fixes #17128

## Before submitting
- [N/A] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Here's the [link](https://github.com/huggingface/transformers/issues/17128)
- [N/A] Did you make sure to update the documentation with your changes? 
- [ ] Did you write any new necessary tests? I didn't write a custom test. Ran the following commands run to ensure local tests pass
1.  `RUN_PIPELINE_TESTS=yes python -m unittest discover -s tests/pipelines  -p "test_pipelines_question_answering.py" -t . -v -f `
2. `python -m unittest discover -s .  -p "test_tokenization_wav2vec2.py" -t . -v -f`

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@LysandreJik
@Narsil

### CODE REVIEW

**Review by Narsil — COMMENTED** [review]
LGTM pinging @LysandreJik !

File: src/transformers/pipelines/question_answering.py [hunk 1]
```diff
@@ -344,7 +339,7 @@ def preprocess(self, example, padding="do_not_pad", doc_stride=None, max_questio
         for i, feature in enumerate(features):
             fw_args = {}
             others = {}
-            model_input_names = self.tokenizer.model_input_names + ["p_mask"]
+            model_input_names = self.tokenizer.model_input_names + ["p_mask", "token_type_ids"]
```
[the remark this answers is not in the document]
Comment (Narsil) [reply src/transformers/pipelines/question_answering.py:342]: There are two ways to pass arguments to the `_forward` function, and the autobatcher will act differently if it's a Tensor or a list. (lists are unbatched, just lists of lists) while tensors get batched (with padded value and direction).

The problem occurs here becaus `token_type_ids` are used in the `postprocessing` (like `p_mask`) so they need to be correctly passed as tensors. So the hint `self.tokenizer.model_input_names` is not enough and this line ensures that the postprocessing will see a tensor aligned with the others.

If `token_type_ids`is listed twice, then maybe we can just (l`model_input_names = set(model_input_names)`  ? (Actually the `fw_args` and `others` are already dictionnary so I think there's only 1 level of duplication.

[the remark this answers is not in the document]
Comment (LysandreJik) [reply src/transformers/pipelines/question_answering.py:342]: Yes that's one solution! Anyway it's a nitpick given that it won't change much. We're using this list to check if the name of a variable is in it, so if it's in it twice, it'll result in the same!

[the remark this answers is not in the document]
Comment (Narsil) [reply src/transformers/pipelines/question_answering.py:342]: You're right!

[the remark this answers is not in the document]
Comment (Narsil) [reply src/transformers/pipelines/question_answering.py:342]: @LysandreJik are you ok with merging this ?

**Review by LysandreJik — APPROVED** [review]
LGTM!

**Comment by LysandreJik** [comment]
Thank you for your contribution, @mygithubid1!

**Pull request merged**